webrtc.html

```diff
@@ -5624,10 +5641,9 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
             </p>
           </li>
           <li>
-            <p>Queue a task to fire an event named <code title=
-            "event-track"><a>track</a></code> with <var>transceiver</var>,
-            <var>track</var>, and <var>streams</var> at the
-            <var>connection</var> object.</p>
+            <p>Add a new <code><a>RTCTrackEvent</a></code> with
+            <var>transceiver</var>, <var>track</var>, and
+            <var>streams</var> to <var>trackEvents</var>.</p>
           </li>
         </ol>
         <p>To <dfn id="process-remote-track-removal">
```

Comment:
In reply to @taylor-b, yes just removing the "queue a task" part from the [step below here](http://w3c.github.io/webrtc-pc/#process-remote-track-removal) that [updates the muted state](https://w3c.github.io/mediacapture-main/getusermedia.html#update-track-muted), seems reasonable to me. Especially since that step in media-capture queues another task itself! That looks like a mistake.

It would even have been nice for consistency perhaps if all `muted` events fired right before all new `track` events, if any, and on the same task as those, but I can't think of a real reason, so that may or may not warrant a change to mediacapture.

That said, the webrtc version we'd remove seems more careful about not firing spurious events, checking in the queued task itself if the state is already in the new state to void firing twice. Perhaps media-capture should do the same at least? 

As a general comment on media-capture, the event-firing algorithms it exposes seem to assume they're being called off-main-thread, or that queuing a separate task for each event firing is desirable. Here it seems we are instead preferring to do multiple events on the same task, to convey a larger set of changes happening as part of the same cause.

Would it have been more flexible if media-capture exposed algorithms that didn't queue a task as their first step, and instead left queuing up to callers of those algorithms? @domenic do you know what common practice is here, and if there is any general spec design advice on this?

Reply:
This algorithm isn't actually invoked by anything in mediacapture-main other than this prose:

> There can be several reasons for a MediaStreamTrack to be muted: the user pushing a physical mute button on the microphone, the user toggling a control in the operating system, the user clicking a mute button in the browser chrome, the User Agent (on behalf of the user) mutes, etc.

Is there any precedent for saying something like "run the following steps, queuing a task to do so if not already running from the relevant event loop"?

Reply:
That, or maybe something like:

"If either of these occur, the User Agent MUST queue a task to **update a track's muted state** to *newState*, using the [user interaction task source](http://w3c.github.io/html/webappapis.html#user-interaction-task-source).

To **update a track's muted state** to *newState*, run the following steps:" ?

This split would let webrtc specify its own task source.

Reply:
Filed a new issue for the "mute" ordering specifically: https://github.com/w3c/webrtc-pc/issues/1568